### PR TITLE
REST API service for Bank Analytics

### PR DESCRIPTION
**Pull request by toplenboren**
Here is swagger doc for the service:

You can check it in https://editor.swagger.io

```(json)
{
    "openapi": "3.0.0",
    "info": {
        "title": "Doma.ai REST API}",
        "version": "1"
    },
    "servers": [
        {
            "url": "https://restapi.d.doma.ai"
        }
    ],
    "paths": {
        "/api/bank-analytics/v1/ticket/all": {
            "get": {
                "parameters": [
                    {
                        "name": "first",
                        "in": "query",
                        "description": "Number of items to be shown (useful for pagination)",
                        "required": true,
                        "schema": {
                            "type": "integer",
                            "minimum": 0
                        }
                    },
                    {
                        "name": "skip",
                        "in": "query",
                        "description": "Number of items to skip (from the last item by sortby query - useful for pagination)",
                        "required": true,
                        "schema": {
                            "type": "integer",
                            "minimum": 0
                        }
                    },
                    {
                        "name": "createdAt_lte",
                        "in": "query",
                        "description": "CreatedAt less then / equal. Should be ISO string",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "createdAt_gte",
                        "in": "query",
                        "description": "CreatedAt greater then / equal. Should be ISO string",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "tin",
                        "in": "query",
                        "description": "TIN of Organization. (INN In Russia)",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "number": {
                                            "type": "integer"
                                        },
                                        "createdAt": {
                                            "type": "string"
                                        },
                                        "propertyAddress": {
                                            "type": "string"
                                        },
                                        "status": {
                                            "type": "object",
                                            "properties": {
                                                "type": {
                                                    "type": "string"
                                                },
                                                "name": {
                                                    "type": "string"
                                                }
                                            }
                                        },
                                        "isEmergency": {
                                            "type": "boolean"
                                        },
                                        "details": {
                                            "type": "string"
                                        },
                                        "assignee": {
                                            "type": "object",
                                            "nullable": true,
                                            "properties": {
                                                "name": {
                                                    "type": "string"
                                                }
                                            }
                                        },
                                        "executor": {
                                            "type": "object",
                                            "nullable": true,
                                            "properties": {
                                                "name": {
                                                    "type": "string"
                                                }
                                            }
                                        },
                                        "unitName": {
                                            "nullable": true,
                                            "type": "string"
                                        },
                                        "unitType": {
                                            "nullable": true,
                                            "type": "string"
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Bad input. You have probably made some mistakes in query params"
                    }
                }
            }
        },
        "/api/bank-analytics/v1/ticket/count": {
            "get": {
                "parameters": [
                    {
                        "name": "status__type",
                        "in": "query",
                        "description": "Status type of ticket. Check condo API for reference",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "createdAt_lte",
                        "in": "query",
                        "description": "CreatedAt less then / equal. Should be ISO string",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "createdAt_gte",
                        "in": "query",
                        "description": "CreatedAt greater then / equal. Should be ISO string",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "tin",
                        "in": "query",
                        "description": "TIN of Organization. (INN In Russia)",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "count": {
                                            "type": "number"
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Bad input. You have probably made some mistakes in query params"
                    }
                }
            }
        },
        "/api/bank-analytics/v1/payment/sum/period": {
            "get": {
                "parameters": [
                    {
                        "name": "period",
                        "in": "query",
                        "description": "Period of ticket. Check condo API for reference. Should be like this: YYYY-MM-01",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "tin",
                        "in": "query",
                        "description": "TIN of Organization. (INN In Russia)",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "sum": {
                                            "type": "number"
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Bad input. You have probably made some mistakes in query params"
                    }
                }
            }
        },
        "/api/bank-analytics/v1/payment/sum/advancedAt": {
            "get": {
                "parameters": [
                    {
                        "name": "advancedAt_lte",
                        "in": "query",
                        "description": "advancedAt less then / equal. Should be ISO string",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "advancedAt_gte",
                        "in": "query",
                        "description": "advancedAt greater then / equal. Should be ISO string",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "tin",
                        "in": "query",
                        "description": "TIN of Organization. (INN In Russia)",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "sum": {
                                            "type": "number"
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Bad input. You have probably made some mistakes in query params"
                    }
                }
            }
        },
        "/api/bank-analytics/v1/billingReceipt/sum/period": {
            "get": {
                "parameters": [
                    {
                        "name": "period",
                        "in": "query",
                        "description": "Period of ticket. Check condo API for reference. Should be like this: YYYY-MM-01",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "name": "tin",
                        "in": "query",
                        "description": "TIN of Organization. (INN In Russia)",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "sum": {
                                            "type": "number"
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Bad input. You have probably made some mistakes in query params"
                    }
                }
            }
        }
    }
}
```

### CODE REVIEW

File: apps/condo-restapi
Comment (sitozzz) [inline apps/condo-restapi:1]: Please provide default app port as 3000

Comment (sitozzz) [inline apps/condo-restapi:1]: What account will be used to access this data?

Comment (sitozzz) [inline apps/condo-restapi:1]: Any restrictions of period usage?

Comment (toplenboren) [reply apps/condo-restapi:1]: Service account with necessary restrictions on PROD

On .d we will use my support account as a temporary solution

I need to unblock guys from big bank :-)

Comment (toplenboren) [reply apps/condo-restapi:1]: Any restrictions will be imposed on condo side. This little lad will just check that params are in good shape.

Comment (toplenboren) [reply apps/condo-restapi:1]: Ok! Done

**Review by VKislov — APPROVED** [review]

**Review by sitozzz — APPROVED** [review]

**Pull request merged**